### PR TITLE
PER, Risk improvements

### PR DESCRIPTION
**Pull request by frozenhelium**

### CODE REVIEW

File: src/components/domain/RiskImminentEvents/Gdacs/EventDetails/index.tsx [hunk 1]
```diff
@@ -1,38 +1,96 @@
...
+        exposed_population,
+        people_affected,
+        impact,
+    } = (exposure?.population_exposure as GdacsPopulationExposure | undefined) ?? {};
+
+    const {
+        url,
+        severitydata,
+        source,
+    } = (event_details as GdacsEventDetails | undefined) ?? {};
```
Comment (tnagorra) [inline src/components/domain/RiskImminentEvents/Gdacs/EventDetails/index.tsx]: Let's not use {} as default value

File: src/components/domain/RiskImminentEvents/Gdacs/index.tsx [hunk 2]
```diff
@@ -142,14 +112,45 @@ function Gdacs(props: Props) {
...
+            }
+
+            const { footprint_geojson } = exposure;
+
+            if (isNotDefined(footprint_geojson)) {
+                return undefined;
+            }
+
+            const footprint = isValidFeatureCollection(footprint_geojson)
+                ? footprint_geojson : undefined;
```
Comment (tnagorra) [inline src/components/domain/RiskImminentEvents/Gdacs/index.tsx]:
```suggestion
            const footprint = isValidFeatureCollection(footprint_geojson)
                ? footprint_geojson
                : undefined;
```

File: src/components/domain/RiskImminentEvents/MeteoSwiss/index.tsx [hunk 3]
```diff
@@ -149,14 +116,64 @@ function MeteoSwiss(props: Props) {
...
+
+            // FIXME: typings should be fixed in the server
+            const {
+                footprint_geojson: {
+                    footprint_geojson,
+                } = {},
+            } = exposure as unknown as { footprint_geojson?: {
+                    footprint_geojson: unknown
+                }
+            };
```
Comment (tnagorra) [inline src/components/domain/RiskImminentEvents/MeteoSwiss/index.tsx]: Let's not use {} as default type

File: src/components/domain/RiskImminentEvents/MeteoSwiss/index.tsx [hunk 4]
```diff
@@ -149,14 +116,64 @@ function MeteoSwiss(props: Props) {
...
+
+            if (isNotDefined(footprint_geojson)) {
+                return undefined;
+            }
+
+            const footprint = isValidFeatureCollection(footprint_geojson)
+                ? footprint_geojson : undefined;
```
Comment (tnagorra) [inline src/components/domain/RiskImminentEvents/MeteoSwiss/index.tsx]:
```suggestion
            const footprint = isValidFeatureCollection(footprint_geojson)
                ? footprint_geojson
                : undefined;
```

File: src/components/domain/RiskImminentEvents/Pdc/index.tsx [hunk 5]
```diff
@@ -59,28 +59,76 @@ function Pdc(props: Props) {
...
+                id,
+                latitude,
+                longitude,
+                hazard_type,
+            } = event;
+
+            if (isNotDefined(latitude)
+                || isNotDefined(longitude)
+                || isNotDefined(hazard_type)
+                || hazard_type === ''
```
Comment (tnagorra) [inline src/components/domain/RiskImminentEvents/Pdc/index.tsx:86]: Let's use isFalsyString

File: src/components/domain/RiskImminentEvents/Pdc/index.tsx [hunk 6]
```diff
@@ -59,28 +59,76 @@ function Pdc(props: Props) {
...
             // FIXME: typings should be fixed in the server
             const stormPositions = (storm_position_geojson as unknown as unknown[] | undefined)
                 ?.filter(isValidPointFeature);
 
+            // forecast_date_time : "2023 SEP 04, 00:00Z"
+            // severity : "WARNING"
+            // storm_name : "HAIKUI"
+            // track_heading : "WNW"
+            // wind_speed_mph : 75
+
```
Comment (tnagorra) [inline src/components/domain/RiskImminentEvents/Pdc/index.tsx:131]: Any reason we need this?

File: src/views/CountryPreparedness/PreviousAssessmentChart/index.tsx [hunk 7]
```diff
@@ -27,6 +29,7 @@ function PreviousAssessmentCharts(props: Props) {
         containerRef,
         {
             xValueSelector: (datum) => datum.assessment_number,
+            xAxisLabelFormatter: (datum) => `Cycle ${datum.assessment_number} (${formatDate(datum.date_of_assessment, 'yyyy')})`,
```
Comment (tnagorra) [inline src/views/CountryPreparedness/PreviousAssessmentChart/index.tsx:33]: Let's translate this

File: src/views/Country/index.tsx [hunk 8]
```diff
@@ -72,13 +74,29 @@ export function Component() {
...
+    );
+
+    if (isDefined(countryResponseError)) {
+        return (
+            <Page
+                title={pageTitle}
+                className={styles.country}
+            >
+                <Message
+                    title="Error loading data for the country!"
```
Comment (tnagorra) [inline src/views/Country/index.tsx]: Let's translate this

File: src/components/domain/RiskImminentEvents/WfpAdam/index.tsx [hunk 9]
```diff
@@ -145,14 +118,45 @@ function WfpAdam(props: Props) {
...
+            }
+
+            const { storm_position_geojson } = exposure;
+
+            if (isNotDefined(storm_position_geojson)) {
+                return undefined;
+            }
+
+            const stormPositions = isValidFeatureCollection(storm_position_geojson)
+                ? storm_position_geojson : undefined;
```
Comment (tnagorra) [inline src/components/domain/RiskImminentEvents/WfpAdam/index.tsx]:
```suggestion
            const stormPositions = isValidFeatureCollection(storm_position_geojson)
                ? storm_position_geojson
                : undefined;
```

File: src/components/domain/RiskImminentEvents/WfpAdam/EventDetails/index.tsx [hunk 10]
```diff
@@ -1,38 +1,161 @@
...
+type WfpAdamItem = NonNullable<WfpAdamResponse['results']>[number];
+type WfpAdamExposure = RiskApiResponse<'/api/v1/adam-exposure/{id}/exposure/'>;
+
+interface WfpAdamPopulationExposure {
+    exposure_60_kmh?: number;
+    exposure_90_kmh?: number;
+    exposure_120_kmh?: number;
+}
+
+interface WfpAdamEventeDetails {
```
Comment (tnagorra) [inline src/components/domain/RiskImminentEvents/WfpAdam/EventDetails/index.tsx]:
```suggestion
interface WfpAdamEventDetails {
```

File: src/components/domain/RiskImminentEvents/WfpAdam/EventDetails/index.tsx [hunk 11]
```diff
@@ -1,38 +1,161 @@
...
+    const stormPoints = useMemo(
+        () => {
+            if (isNotDefined(exposure)) {
+                return undefined;
+            }
+
+            const { storm_position_geojson } = exposure;
+
+            const geoJson = isValidFeatureCollection(storm_position_geojson)
+                ? storm_position_geojson : undefined;
```
Comment (tnagorra) [inline src/components/domain/RiskImminentEvents/WfpAdam/EventDetails/index.tsx]:
```suggestion
            const geoJson = isValidFeatureCollection(storm_position_geojson)
                ? storm_position_geojson
                : undefined;
```

File: src/components/domain/RiskImminentEvents/WfpAdam/EventDetails/index.tsx [hunk 12]
```diff
@@ -1,38 +1,161 @@
...
+
+            const points = geoJson?.features.map(
+                (pointFeature) => {
+                    if (!isValidPointFeature(pointFeature)
+                        || isNotDefined(pointFeature.properties)
```
Comment (tnagorra) [inline src/components/domain/RiskImminentEvents/WfpAdam/EventDetails/index.tsx]:
```suggestion
                    if (
                        !isValidPointFeature(pointFeature)
                        || isNotDefined(pointFeature.properties)
```

File: src/components/domain/RiskImminentEvents/WfpAdam/EventDetails/index.tsx [hunk 13]
```diff
@@ -1,38 +1,161 @@
...
+        effective_date,
+        dashboard_url,
+        alert_level,
+        flood_area,
+        fl_croplnd,
+        source,
+        sitrep,
+        url,
+        mag,
+    } = (event_details as WfpAdamEventeDetails | undefined) ?? {};
```
Comment (tnagorra) [inline src/components/domain/RiskImminentEvents/WfpAdam/EventDetails/index.tsx]: Let's not use {} as default type

File: src/components/domain/RiskImminentEvents/WfpAdam/EventDetails/index.tsx [hunk 14]
```diff
@@ -1,38 +1,161 @@
...
+
+    const {
+        exposure_60_kmh,
+        exposure_90_kmh,
+        exposure_120_kmh,
+    } = (exposure?.population_exposure as WfpAdamPopulationExposure | undefined) ?? {};
+
```
Comment (tnagorra) [inline src/components/domain/RiskImminentEvents/WfpAdam/EventDetails/index.tsx]: Let's not use {} as default type

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: src/components/domain/RiskImminentEvents/Pdc/index.tsx (hunk 5 above)
Comment (frozenhelium) [reply src/components/domain/RiskImminentEvents/Pdc/index.tsx:86]: This is done to fix the typing, using `isFalsyString` would not solve the issue here

File: src/components/domain/RiskImminentEvents/Pdc/index.tsx (hunk 6 above)
Comment (frozenhelium) [reply src/components/domain/RiskImminentEvents/Pdc/index.tsx:131]: Yep, these are the properties in geojson for the reference

**Review by tnagorra — APPROVED** [review]

**Pull request merged**